src/neptune/objects/neptune_object.py

```diff
@@ -343,6 +291,8 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             traceback.print_exception(exc_type, exc_val, exc_tb)
         self.stop()
 
+        super().__exit__(exc_type, exc_val, exc_tb)
```

Comment:
Are you sure that with non-context manager approach are we going to close the backend connection?
```python
run = Run(mode='read-only')
run.stop()
```
With the sample above, `backend.close()` will not be called.